src/viam/sdk/common/exception.hpp

```diff
@@ -0,0 +1,52 @@
+/// @file common/exception.hpp
+///
+/// @brief Defines custom exceptions for the SDK.
+#pragma once
+#include <stdexcept>
+#include <string>
+
+#include <viam/sdk/resource/resource_api.hpp>
+
+namespace viam {
+namespace sdk {
+
+/// @defgroup Exception Classes related to SDK exceptions.
+
+/// @class ViamErrorCode
+/// @brief Defines a set of a error codes to be used in conjunction with
+/// ViamException.
+/// @ingroup Exception
+enum class ViamErrorCode {
+    Unknown = 0,            // Default code
+    Connection,             // Issue during connection establishment
+    DuplicateRegistration,  // API or API/Model pair has already been registered
+    DuplicateResource,      // Resource has already been added
+    GRPC,                   // gRPC error from remote machine
```

Comment:
If we formulated an error category for gRPC's error codes, could we do more?

Reply:
So I decided to leave the `k_grpc` error condition enum but made a new `GRPCException` class that is a child class of `Exception` and carries a private `grpc::Status` field. Let me know what you think.